pkgs/by-name/st/starpu/package.nix

```diff
@@ -0,0 +1,121 @@
+{
+  # derivation dependencies
+  lib,
+  fetchurl,
+  stdenv,
+
+  # starpu dependencies
+  hwloc,
+  libuuid,
+  libX11,
+  fftw,
+  fftwFloat, # Same than previous but with float precision
+  pkg-config,
+  libtool,
+  autoconf,
+  automake,
+  simgrid,
+  mpi,
+  cudaPackages,
+
+  # Options
+  enableSimgrid ? false,
+  enableMPI ? false,
+  enableCUDA ? false,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "starpu";
+  version = "1.4.7";
+
+  inherit enableSimgrid;
+  inherit enableMPI;
+  inherit enableCUDA;
+
+  src = fetchurl {
+    url = "https://files.inria.fr/starpu/starpu-${finalAttrs.version}/starpu-${finalAttrs.version}.tar.gz";
+    hash = "sha256-HrPfVRCJFT/m4LFyrZURhDS0qB6p6qWiw4cl0NtTsT4=";
+  };
+
+  # Runtime build dependencies
+  nativeBuildInputs =
+    [
+      pkg-config
+      hwloc
+    ]
+    ++ lib.optional finalAttrs.enableSimgrid simgrid
+    ++ lib.optional finalAttrs.enableMPI mpi
+    ++ lib.optional finalAttrs.enableCUDA cudaPackages.cudatoolkit;
+
+  buildInputs =
+    [
+      libuuid
+      libX11
+      fftw
+      fftwFloat
+      pkg-config
+      libtool
+      autoconf
+      automake
```

Comment:
Most likely these four (pkg-config, libtool, autoconf/automake) need to go in nativeBuildInputs.